includes/Core/Authentication/Authentication.php

```diff
@@ -1404,13 +1399,12 @@ private function filter_features_via_proxy( $feature_enabled, $feature_name ) {
 			}
 
 			if ( ! $this->credentials->has() ) {
-				// For the 'serviceSetupV2' feature, continue to check for the legacy persistent option which used to
-				// be set when remote features were cached as transients.
-				if ( 'serviceSetupV2' === $feature_name && $this->options->get( $service_setup_v2_option_name ) ) {
+				// Use a persistent option so that it remains active even if the
+				// site is not connected. This is crucial to provide a
+				// consistent setup flow experience.
+				if ( $this->options->get( $service_setup_v2_option_name ) ) {
 					return true;
 				}
```

Comment:
I think the option is still no longer necessary here because the feature is on 100% of the time now. As is, this would enable all features if the user had this option set which is not what we want 😄 

The important thing to keep here is that we don't attempt to fetch remote features if there are no credentials as that request would fail.

I'm about to push the fixes for user input now so I'll fix this as well 👍

Reply:
Cheers, I think after the `main` merge this code sorta confused me 😅